recipes/gdal/all/conanfile.py

```diff
@@ -359,7 +353,7 @@ def _validate_dependency_graph(self):
 
     @property
     def _settings_build(self):
```

Comment:
Question:
In `validate`, there is a call to `tools.cross_building`.
I think the argument to it should be `self` instead of `self.settings`.

Reply:
most of the time, I see tools.cross_building(self.settings) in CCI recipes (specially test package), is it really important?

Reply:
it should be `tools.cross_building(self)` instead of `tools.cross_building(self.settings)`
for two profiles case, at least, it needs to read `settings_build` and `settings_host` from the `conanfile` object.

Reply:
It would make sense.
But `tools.cross_building(self.settings)` returns True with 2 profiles.

Reply:
it has two code paths, basically
it checks if first arguments is an instance of `ConanFile`. if so, it uses `conanfile.settings_build`
otherwise, it uses `arch_build` and `os_build` instead
it may work (e.g. if you have also set arch_build/os_build), but it's not guaranteed.
in conan 2.0 (and in conan v2 mode) it is likely the helper will not accept `Settings` as a first argument.
there is even a warning right now:
```
    if settings:
        warnings.warn("Argument 'settings' has been deprecated, use 'conanfile' instead")
```

Reply:
yes, it's a time. you can submit PR, if you want, otherwise me or @uilianries will take care about that
https://github.com/conan-io/hooks/issues/324

Reply:
And about tools.cross_building in validate()... it is always True in CI of CCI on the Linux machine evaluating package id, so none of the macOS & Visual Studio shared configurations have been generated :(